DSCResources/MSFT_xVMDvdDrive/MSFT_xVMDvdDrive.psm1

```diff
@@ -0,0 +1,428 @@
+#region localizeddata
+if (Test-Path "${PSScriptRoot}\${PSUICulture}")
+{
+    Import-LocalizedData `
+        -BindingVariable LocalizedData `
+        -Filename MSFT_xVMDvdDrive.strings.psd1 `
+        -BaseDirectory "${PSScriptRoot}\${PSUICulture}"
+}
+else
+{
+    #fallback to en-US
+    Import-LocalizedData `
+        -BindingVariable LocalizedData `
+        -Filename MSFT_xVMDvdDrive.strings.psd1 `
+        -BaseDirectory "${PSScriptRoot}\en-US"
+}
+#endregion
+
+# Import the common HyperV functions
+Import-Module -Name ( Join-Path `
+    -Path (Split-Path -Path $PSScriptRoot -Parent) `
+    -ChildPath '\HyperVCommon\HyperVCommon.psm1' )
+
+<#
+    .SYNOPSIS
+    Returns the current status of the VM DVD Drive.
+
+    .PARAMETER VMName
+    Specifies the name of the virtual machine to which the DVD drive is to be added.
+
+    .PARAMETER ControllerNumber
+    Specifies the number of the controller to which the DVD drive is to be added.
+
+    .PARAMETER ControllerLocation
+    Specifies the number of the location on the controller at which the DVD drive is to be added.
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $VMName,
+
+        [parameter(Mandatory = $true)]
+        [System.Uint32]
+        $ControllerNumber,
+
+        [parameter(Mandatory = $true)]
+        [System.Uint32]
+        $ControllerLocation
+    )
+
+    Write-Verbose -Message ( @(
+            "$($MyInvocation.MyCommand): "
+            $($LocalizedData.GettingVMDVDDriveMessage `
+                -f $VMName,$ControllerNumber,$ControllerLocation)
+        ) -join '' )
+
+    Test-ParameterValid @PSBoundParameters
+
+    $dvdDrive = Get-VMDvdDrive @PSBoundParameters
```

Comment:
Should we not assert that Hyper-V/the cmdlets are installed and throw a nice error message (as per other resource modules)?